clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -566,6 +565,8 @@ private List<ProducerBatch> drainBatchesForOneNode(Cluster cluster, Node node, i
         do {
             PartitionInfo part = parts.get(drainIndex);
             TopicPartition tp = new TopicPartition(part.topic(), part.partition());
+            updateDrainIndex(node.idString(), drainIndex);
+            drainIndex = (drainIndex + 1) % parts.size();
```

Comment:
update the drainIndex  before increase drainIndex is necessary

Reply:
Thanks, and please add a test for the `continue` case. Thanks.